modules/openapi-generator/src/main/resources/csharp-netcore/modelGeneric.mustache

```diff
@@ -300,6 +300,7 @@
             {{/parent}}
             {{#vars}}
             {{#hasValidation}}
+            {{^isEnum}}
             {{#maxLength}}
             // {{{name}}} ({{{dataType}}}) maxLength
             if(this.{{{name}}} != null && this.{{{name}}}.Length > {{maxLength}})
```

Comment:
Instead of not validating for enum params. What about if:
`this.{{{name}}}.ToString().Length > {{maxLength}}` so that it will work for other case like:

```
	uuid_property:
      description: This is description
      type: string
      format: uuid
      maxLength: 36
```

Else, it will also break at: `UuidProperty.Length > 36`

Reply:
I'm not sure I understand what you mean, are you saying strings with `format: uuid` are not String objects and thus need the `toString()` call? Looks like a separate issue if that's what you meant.

Here the fix simply skips adding validation checks if the property specifies an enum, which is a very reliable and future-proof way, me thinks.

Reply:
Strings with format: uuid are of Guid type in c#. The comment is valid however and worth some thought. If there is a simple way to keep validation and still have the enum behavior that might be worth exploring. However having validation for enum parameters would mean that they might break at some point in the future when one decides to use the forbidden values for the first time instead of breaking immediately.

Reply:
Yes. I meant to say is `Strings with format: uuid are of Guid type in c#`.
And, as the above validations like `MaxLength`/ `MinLength` are specific to `string` type. Its better to convert to string type using `ToString()` before validating the property.
Similar to that I found in `ruby-client` -> https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/ruby-client/partial_model_generic.mustache#L302 ( `to_s` converts to `string` IIUC)

Besides, it might be separate issue but the gist of the bug is same.

Reply:
Guids really don't have the same issue however. Reason being that it performs its own validation. Of course the same applies you could technically have the validation parameters in the spec I guess. It just makes no sense whatsoever.